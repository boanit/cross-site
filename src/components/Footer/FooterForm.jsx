/*
 * @Author: Kenneth Kwakye-Gyamfi
 * @Date: 2019-04-11 21:30:39
 * @Last Modified by: Kenneth Kwakye-Gyamfi
 * @Last Modified time: 2019-04-11 21:38:12
 */
import React from 'react';

export default class FooterForm extends React.Component {
	state = {
		name: '',
		email: '',
		message: ''
	};

	render() {
		return (
			<form
				action="#"
				method="post"
				name="contact"
				className="footer-form"
				data-netlify="true">
				<input
					id="name"
					type="text"
					name="name"
					placeholder="Name"
					value={this.state.name}
					className="footer-form__input"
					onChange={e => this.setState({ name: e.target.value })}
				/>

				<input
					id="email"
					name="email"
					type="email"
					className="footer-form__input"
					value={this.state.email}
					onChange={e => this.setState({ email: e.target.value })}
					placeholder="Email address"
				/>

				<textarea
					id="message"
					name="message"
					className="footer-form__input footer-form__textarea"
					value={this.state.message}
					onChange={e => this.setState({ message: e.target.value })}
					placeholder="Your message"
				/>

				<button type="submit" className="footer-form__button">
					Submit
				</button>
			</form>
		);
	}
}
