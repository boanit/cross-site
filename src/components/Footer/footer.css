/*
 * @Author: Kenneth Kwakye-Gyamfi
 * @Date: 2019-04-11 21:19:21
 * @Last Modified by: Kenneth Kwakye-Gyamfi
 * @Last Modified time: 2019-04-11 21:51:58
 */
.footer {
	background-color: #252934;
	position: relative;
	padding: 2.5rem 1rem 0;
	min-height: 20rem;
}

@media screen and (min-width: 40rem) {
	.footer {
		padding: 10rem 1rem 0;
		-webkit-clip-path: polygon(50% 5rem, 100% 0, 100% 100%, 0 100%, 0 0);
		clip-path: polygon(50% 5rem, 100% 0, 100% 100%, 0 100%, 0 0);
	}
}

.footer__title {
	text-align: center;
	line-height: 1.4;
	color: #ffffff;
	font-size: 1.75rem;
	margin-bottom: 1rem;
}

.footer__subtitle {
	color: #009688;
	text-align: center;
	line-height: 1.4;
	font-size: 1.25rem;
	margin-bottom: 2rem;
}

/*#region Footer form */
.footer-form {
	width: 100%;
	margin: 0 auto;
	display: flex;
	flex-direction: column;
	align-items: flex-end;
	justify-content: flex-start;
	margin-bottom: 7.5rem;
}

@media screen and (min-width: 30rem) {
	.footer-form {
		width: 30rem;
	}
}

.footer-form__input {
	width: 100%;
	margin-bottom: 1rem;
	background: #1e242c;
	border: none;
	padding: 1rem;
	color: #ffffff;
	font-size: 0.9rem;
}

.footer-form__textarea {
	min-height: 10rem;
	resize: vertical;
}

.footer-form__button {
	padding: 0.75rem 1.5rem;
	background: transparent;
	border: 1px solid #ffffff;
	font-size: 0.9rem;
	color: #ffffff;
}
/*#endregion*/

.footer__base {
	padding: 3rem 1rem;
	background-color: #1b242f;
	width: calc(100% + 2rem);
	margin: 0 -1rem;
	position: relative;
}

.footer-base__scroll-to {
	position: absolute;
	width: 3rem;
	height: 3rem;
	border: 1px solid #ffffff;
	background-color: transparent;
	top: -1.5rem;
	left: calc(50% - 1.5rem);
	color: #ffffff;
	transition: 175ms;
	cursor: pointer;
}

.footer-base__scroll-to:hover,
.footer-base__scroll-to:active {
	background-color: #ffffff;
	color: #1b242f;
}

.copyright {
	color: #aaaaaa;
	text-align: center;
	margin-bottom: 0.5rem;
	line-height: 1.4;
}

.footer-base__github {
	text-align: center;
	line-height: 1.4;
	color: #bbbbbb;
}

.footer-base__github a {
	color: #eeeeee;
	border-bottom: 1px dotted #eeeeee;
}

.footer-base__github a i {
	display: inline-block;
	margin-right: 0.5rem;
}
