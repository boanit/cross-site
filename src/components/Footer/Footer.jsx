/*
 * @Author: Kenneth Kwakye-Gyamfi
 * @Date: 2019-04-11 21:17:16
 * @Last Modified by: Kenneth Kwakye-Gyamfi
 * @Last Modified time: 2019-04-11 21:50:21
 */
import React from 'react';
import { animateScroll as scroll } from 'react-scroll';

import FooterForm from './FooterForm';
import './footer.css';

function Footer() {
	return (
		<footer className="footer">
			<h2 className="footer__title">Contact</h2>
			<p className="footer__subtitle">Have a question or want us to work together?</p>

			<FooterForm />

			<div className="footer__base">
				<button
					onClick={scroll.scrollToTop}
					className="footer-base__scroll-to"
					title="Click to scroll to the top">
					<i className="ti-angle-double-up" />
				</button>
				<p className="copyright">Kwakye-Gyamfi Kenneth &copy; 2019</p>
				<p className="footer-base__github">
					View this project on{' '}
					<a href="https://www.github.com/cr05s19xx/cross-site">
						<i className="ti-github" />
						<span>Github</span>
					</a>
				</p>
			</div>
		</footer>
	);
}

export default React.memo(Footer);
