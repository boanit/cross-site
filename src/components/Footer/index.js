/*
 * @Author: Kenneth Kwakye-Gyamfi
 * @Date: 2019-04-11 21:17:11
 * @Last Modified by: Kenneth Kwakye-Gyamfi
 * @Last Modified time: 2019-04-11 21:17:49
 */
import Footer from './Footer';

export default Footer;
