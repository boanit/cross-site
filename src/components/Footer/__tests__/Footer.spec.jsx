/*
 * @Author: Kenneth Kwakye-Gyamfi
 * @Date: 2019-04-28 22:17:53
 * @Last Modified by: Kenneth Kwakye-Gyamfi
 * @Last Modified time: 2019-04-28 22:19:50
 */
import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import Footer from '../Footer';

describe('Component Testing', () => {
	const component = <Footer />;

	test('renders without crashing', () => {
		const wrapper = shallow(component);
		expect(wrapper.exists()).toBeTruthy();
	});

	test('matches snapshot', () => {
		const snapshot = create(component).toJSON();
		expect(snapshot).toMatchSnapshot();
	});
});
