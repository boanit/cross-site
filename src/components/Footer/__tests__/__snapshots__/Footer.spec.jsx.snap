// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Testing matches snapshot 1`] = `
<footer
  className="footer"
>
  <h2
    className="footer__title"
  >
    Contact
  </h2>
  <p
    className="footer__subtitle"
  >
    Have a question or want us to work together?
  </p>
  <form
    action="#"
    className="footer-form"
    data-netlify="true"
    method="post"
    name="contact"
    onSubmit={[Function]}
  >
    <input
      className="footer-form__input"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      className="footer-form__input"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email address"
      type="email"
      value=""
    />
    <textarea
      className="footer-form__input footer-form__textarea"
      id="message"
      name="message"
      onChange={[Function]}
      placeholder="Your message"
      value=""
    />
    <button
      className="footer-form__button"
      type="submit"
    >
      Submit
    </button>
  </form>
  <div
    className="footer__base"
  >
    <button
      className="footer-base__scroll-to"
      onClick={[Function]}
      title="Click to scroll to the top"
    >
      <i
        className="ti-angle-double-up"
      />
    </button>
    <p
      className="copyright"
    >
      Kwakye-Gyamfi Kenneth Â© 2019
    </p>
    <p
      className="footer-base__github"
    >
      View this project on
       
      <a
        href="https://www.github.com/cr05s19xx/cross-site"
      >
        <i
          className="ti-github"
        />
        <span>
          Github
        </span>
      </a>
    </p>
  </div>
</footer>
`;
